lib/portage/sync/syncbase.py

```diff
@@ -303,6 +300,26 @@ def noisy_refresh_keys():
 				loop.run_until_complete(decorated_func())
 		out.eend(0)
 
+	def _get_openpgp_env(self, system=None):
```

Comment:
@mgorny what about this new attempt?

Reply:
Instead of `system=`, you could pass the path to the key file and return system env when `None` (returning `None` vs system env is not very important). You could also put key loading and maybe refresh in the common function since it seems to be the same everywhere.